File: app/src/main/java/com/akameko/testforupstarts/MainFragment.java
package com.akameko.testforupstarts;

import androidx.lifecycle.ViewModelProvider;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import androidx.room.Room;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;

import com.akameko.testforupstarts.repository.pojos.Jeans;
import com.akameko.testforupstarts.repository.retrofit.Repository;
import com.akameko.testforupstarts.repository.room.JeansDatabase;

import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.Disposable;
import io.reactivex.schedulers.Schedulers;

public class MainFragment extends Fragment {

    private SharedViewModel sharedViewModel;

    public static MainFragment newInstance() {
        return new MainFragment();
    }



    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View root =inflater.inflate(R.layout.main_fragment, container, false);
        sharedViewModel = new ViewModelProvider(requireActivity()).get(SharedViewModel.class);
        initFragment();
        return root;
    }

    private  void initFragment(){
        sharedViewModel.loadJeans();
        sharedViewModel.jeansList.observe(getViewLifecycleOwner(), jeansList -> {
                initRecycler(jeansList);
        });
    }



    public void initRecycler(List<Jeans> jeansList) {

        RecyclerView recyclerView = getActivity().findViewById(R.id.main_recycler_view);
        recyclerView.setItemAnimator(new DefaultItemAnimator());
        recyclerView.setHasFixedSize(true);

        GridLayoutManager layoutManager = new GridLayoutManager(getContext(),2);
        recyclerView.setLayoutManager(layoutManager);

        List<Jeans> liked = sharedViewModel.jeansDatabase.getJeansDao().getAllItems();
        MainAdapter mainAdapter = new MainAdapter(jeansList, liked);

        mainAdapter.setOnLikeClickListener((likedJeans, position, addToFavourite) -> {
            if (addToFavourite){
                sharedViewModel.jeansDatabase.getJeansDao().insert(likedJeans);
                Log.d("123", "Добавлено в избранное");
                //Log.d("123", jeansDatabase.getJeansDao().getItemById(likedJeans.getId()).toString());
            } else {
                sharedViewModel.jeansDatabase.getJeansDao().delete(likedJeans);
                Log.d("123", "Убрано из избранного");
            }


        });

        mainAdapter.setOnItemClickListener((jeansToShow, position) -> {
            sharedViewModel.setActiveJeans(jeansToShow);
            sharedViewModel.setPositionToShow(position);
            ((MainActivity)getActivity()).showDetails();
        });

        recyclerView.setAdapter(mainAdapter);
    }

}

File: app/src/main/java/com/akameko/testforupstarts/MainFragment.kt
package com.akameko.testforupstarts

import android.os.Bundle
import android.os.Handler
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.akameko.testforupstarts.repository.pojos.Jeans

class MainFragment : Fragment() {
    private var sharedViewModel: SharedViewModel? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val root = inflater.inflate(R.layout.main_fragment, container, false)
        sharedViewModel = ViewModelProvider(requireActivity()).get(SharedViewModel::class.java)
        initFragment()
        return root
    }

    private fun initFragment() {
        sharedViewModel!!.loadJeans()
        sharedViewModel!!.jeansList.observe(viewLifecycleOwner, Observer { jeansList: List<Jeans?> -> initRecycler(jeansList) })
    }

    fun initRecycler(jeansList: List<Jeans?>) {
        val recyclerView: RecyclerView = activity!!.findViewById(R.id.main_recycler_view)
        recyclerView.itemAnimator = DefaultItemAnimator()
        recyclerView.setHasFixedSize(true)
        val layoutManager = GridLayoutManager(context, 2)
        recyclerView.layoutManager = layoutManager
        val liked = sharedViewModel!!.jeansDatabase!!.jeansDao.allItems
        val mainAdapter = MainAdapter(jeansList, liked)
        mainAdapter.setOnLikeClickListener { likedJeans: Jeans?, position: Int, addToFavourite: Boolean ->
            if (addToFavourite) {
                sharedViewModel!!.jeansDatabase!!.jeansDao.insert(likedJeans)
                Log.d("123", "Добавлено в избранное")
                showLikeNotification()
                //Log.d("123", jeansDatabase.getJeansDao().getItemById(likedJeans.getId()).toString());
            } else {
                sharedViewModel!!.jeansDatabase!!.jeansDao.delete(likedJeans)
                Log.d("123", "Убрано из избранного")
            }
        }
        mainAdapter.setOnItemClickListener { jeansToShow: Jeans?, position: Int ->
            sharedViewModel!!.activeJeans = jeansToShow
            sharedViewModel!!.positionToShow = position
            (activity as MainActivity?)!!.showDetails()
        }
        recyclerView.adapter = mainAdapter
        val textViewCounter = activity!!.findViewById<TextView>(R.id.text_view_main_count)
        textViewCounter.text = String.format(getString(R.string.text_counter), jeansList.size)
    }

    private fun showLikeNotification() {
        val cardViewNotification: CardView = activity!!.findViewById(R.id.card_view_notification_main)
        cardViewNotification.visibility = View.VISIBLE
        val h = Handler()
        h.postDelayed({ cardViewNotification.visibility = View.GONE }, 3000)
    }

    companion object {
        fun newInstance(): MainFragment {
            return MainFragment()
        }
    }
}